src/mbgl/renderer/render_source.cpp

```diff
@@ -64,12 +64,39 @@ void RenderSource::setObserver(RenderSourceObserver* observer_) {
 
 void RenderSource::onTileChanged(Tile& tile) {
     observer->onTileChanged(*this, tile.id);
+    if (tile.isComplete()) {
+        const auto tileID = tile.id.toUnwrapped();
+        const auto& idealTile = idealTilesLoaded.find(tileID);
```

Comment:
```
if (tilePyramid.getIdealTiles().count(tileID)) {
idealTilesLoaded.insert(tileID);
}
```

Reply:
Couple issues here. RenderSource does not have visibility to tilePyramid. Did you mean` if (tilePyramid.getIdealTiles().count(tileID) == 0) `? If the tile already exists,  `insert` will not do anything.